src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -28,14 +28,13 @@ private JsonPropertyInfo AddProperty(Type propertyType, PropertyInfo propertyInf
             // Convert non-immutable dictionary interfaces to concrete types.
             if (propertyType.IsInterface && jsonInfo.ClassType == ClassType.Dictionary)
             {
-                // If a polymorphic case, we have to wait until run-time values are processed.
```

Comment:
What was this trying to guard against?

Reply:
See https://github.com/dotnet/corefx/pull/37300/files#r280149022.
It was trying to prevent attempting to (de)serialize `object` types. Now that `JsonElement` is in, we can lift this guard as we need to get a concrete type for `IDictionary<string, object>` (-> `Dictionary<string, object>`)

Reply:
https://github.com/dotnet/corefx/pull/38512#issuecomment-502192249